manticore/core/cpu/abstractcpu.py

```diff
@@ -416,15 +428,15 @@ def canonicalize_instruction_name(self, instruction):
         pass
 
     def execute(self):
-        ''' Decode, and execute one instruction pointed by register PC'''
-        if not isinstance(self.PC, (int,long)):
-            raise SymbolicPCException()
+        '''
+        Decode, and execute one instruction pointed by register PC
+        '''
 
-        if not self.memory.access_ok(self.PC,'x'):
-            raise InvalidPCException(self.PC)
+        # Decode the instruction if it wasn't explicitly decoded
+        if self._decoded_pc != self.PC:
```

Comment:
instead of creating a new class member just for this check, can we use `self.instruction.address` for this? `if self.instruction.address != self.PC`